site/content/api/Command-And-Control-API.md

```diff
@@ -12,66 +12,52 @@ THIS IS A FIRST DRAFT VERSION OF THE API SPEC AND THUS STILL SUBJECT TO DISCUSSI
-2. Client has established an AMQP link in role *receiver* with Hono using target address `control/${tenant_id}/${device_id}`. Where `${device_id}` is the ID of the device to receive control messages for and `${tenant_id}` is the ID of the tenant the device belongs to.
+**Link establishment and Message Flow**
 
-**Message Flow**
+The following sequence diagram shows the whole message exchange including the response being sent back to the Business Application.
 
-*TODO* add sequence diagram
+![Send Command](../command_control_Success.png)
 
 **Message Format**
 
-The following table provides an overview of the properties a client needs to set on a *Control* message.
+The following table provides an overview of the properties the *Business Application* needs to set on a command message.
 
 | Name | Mandatory | Location | Type      | Description |
 | :--- | :-------: | :------- | :-------- | :---------- |
-| *message-id* | yes | *properties* | UTF-8 *string* | MUST contain the message ID used to match incoming control message with the response message containing result of execution. |
-| *device_id* | yes | *application-properties* | UTF-8 *string* | MUST contain the ID of the device the data in the payload has been reported by. |
-| *tenant_id* | yes | *application-properties* | UTF-8 *string* | MAY be set by the client to indicate which tenant the the device belongs to. |
-| *command* | no | *application-properties* | UTF-8 *string* | MUST contain the command to be executed by a device. |
-|*ttl* | no | *properties* | *int* | Indicates the timeout value (in milliseconds) indicating when server can assume that command has not been executed because of the lack of the response from the client. Default timeout is *60000* (1 minute). |
-|*durable* | no | *properties* | *boolean* | Indicates if the message should be persisted (*store and forward*) by the underlying message transport middleware. By default command&control messages are durable.  |
-
-If a control message requires additional data to be properly executed (like command parameters), the body of the message MUST consist of a single AMQP *Data* section containing a control 
-message data. The format and encoding of the data is indicated by the *command*. It is up to the device to interpret the incoming body by taking `command` into 
-consideration.
-
-## Sending back control message execution result
-
-After execution of a control message on a device, device MUST send a response message indicating a status of that execution.
-
-**Preconditions**
+| *subject*        | yes       | *properties*             | *string*     | MUST contain the command name to be executed by a device. |
+| *correlation-id* | no        | *properties*             | *message-id* | MAY contain an ID used to correlate a response message to the original request. If set, it is used as the *correlation-id* property in the response, otherwise the value of the *message-id* property is used. |
```

Comment:
this property has no default value in the request. The request's message-id is used as the correlation-id in the *response*, if no specific *correlation-id* has been set in the request.

Reply:
I do not really understand your point?

Reply:
the table defines the properties that can be contained in a command request, right? If so, then the *correlation-id* has no default value. The request message's *correlation-id* property is empty by default. The device, however, **MUST** set the *correlation-id* property of the response message to the request's *message-id* if the request **did not** contain a *correlation-id*.

Reply:
the row for the *correlation-id* contains *message-id* in the fourth column, which is used to indicate the default value if not set explicitly ...

Reply:
No, it is the type column.